changelogs/fragments/254_variable_warnings.yml

```diff
@@ -0,0 +1,5 @@
+---
+minor_changes:
```

Comment:
I think this is breaking change. What do you think @Andersson007 ?

Reply:
@Akasurde i think the warning itself isn't, so i believe that's fine.
If the type change wasn't announced yet, we should add another fragment under `major_changes:` to announce that the change is coming soon.

Reply:
Ok, I think then we can merge this https://github.com/ansible-collections/ansible.posix/pull/249 after this PR is merged.

Reply:
yeah